CenterForOpenScience/osf.io
# [ENG-2334] Make DOIs mint on project/draft registration

## Purpose

This is primarily a ticket for ensuring that DOIs mint of project draft registration, but at some time in the distant past the ability to mint DOIs locally and therefore test this feature was disabled, so I had to make some tweaks in order to restore that behavior.

## Changes

- tweaks client (apparently just flat broken) to allow for minting DOIs and restructure tests to be mocked properly.
- removes old ServiceTemporarilyUnavailable exception
- removes old settings checks that were getting in the way of tests.  

## QA Notes

- Verify at https://doi.test.datacite.org/repositories/demo.cos/dois that DOIs are being minted when a project or draft reg is registered. ⚠️ You will need appropriate credentials to do this. ⚠️ 
- Verify Registration process is timely and error free.

## Documentation

QA: is there any user facing docs about this?

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/ENG-2334
